crates/torii/libp2p/src/server/mod.rs

```diff
@@ -302,37 +302,15 @@ impl<P: Provider + Sync> Relay<P> {
                             // to prevent replay attacks.
 
                             // Verify the signature
-                            let message_hash =
-                                if let Ok(message) = data.message.encode(entity_identity) {
-                                    message
-                                } else {
-                                    info!(
-                                        target: LOG_TARGET,
-                                        "Encoding message."
-                                    );
-                                    continue;
-                                };
-
-                            let mut calldata = vec![message_hash];
-                            calldata.push(Felt::from(data.signature.len()));
-
-                            calldata.extend(data.signature);
-                            if !match self
-                                .provider
-                                .call(
-                                    FunctionCall {
-                                        contract_address: entity_identity,
-                                        entry_point_selector: get_selector_from_name(
-                                            "is_valid_signature",
-                                        )
-                                        .unwrap(),
-                                        calldata,
-                                    },
-                                    BlockId::Tag(BlockTag::Pending),
-                                )
-                                .await
+                            if !match validate_signature(
+                                &self.provider,
+                                entity_identity,
+                                &data.message,
+                                &data.signature,
+                            )
+                            .await
```

Comment:
Since the function returns a bool, the construction of this and arms below may be much simpler:

```rust
if !validate_signature(...)? {
    error!(..);
    continue
}
```

Reply:
this wouldn't work though since we don't want to propagate the error. we should just log it and not exit. since we're taking in user input